### PR TITLE
Fix UnsubscribeButton wrong result code in case of UNSUPPORTED_RESOURCE

### LINKED ISSUE #998: SDL returns IGNORED instead of UNSUPPORTED_RESOURCE for UnsubscribeButton
## Occurrence
Always.

## Preconditions
1. In file hmi_capabilities.json "SEARCH" button exists in capabilities.
2. Core, HMI are started.
3. SDL receives Buttons.GetCapabilities response. "SEARCH" button is not in list for supported buttons.

## Steps to reproduce
1. Send RPC UnsubscribeButton with buttonName = "SEARCH".

## Expected result
SDL shall return result_code: UNSUPPORTED_RESOURCE, success: false

## Actual result
SDL returns result_code: IGNORED, success: false.

**Note:**
Behaviour of SDL for RPC SubscribeButton in case not supported button is correct.

# Environment
Release: 4.2.0 (master marge)
Commit: a6c32b45832653d6b851be9b19355fa35f377e07
OS: Ubuntu 14.04 
Transport: WiFi 
App type: navi
App: Luxoft SPT
Device: HTC One 

# Attachments
[att_issue#998.zip](https://github.com/smartdevicelink/sdl_core/files/576080/att_issue.998.zip)

# Expected delivery   
- [ ] Source code updates 
- [ ] Code comments 
- [ ] UTs add/update 
- [x] ATF tests add/update ***(not required)***  
- [x] Manual tests ***(not required)*** 
- [x] Add/update CI plans/jobs ***(not required)***  
- [x] SDD updates ***(not required)***  
- [x] Guidelines update ([sdl_core_guides](https://github.com/smartdevicelink/sdl_core_guides)) ***(not required)***  
- [x] Guidelines update ([sdl_hmi_integration_guidelines](https://github.com/smartdevicelink/sdl_hmi_integration_guidelines))  ***(not required)***

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixed `UnsubscribeButton` wrong result code in case of `UNSUPPORTED_RESOURCE`.
This checking was already implemented for `SubscribeButton` rpc, but was not implemented for `UnsubscribeButton`.

In this pull request:
- Function `CheckHMICapabilities()`, which responsible for HMI capabilities permissions checking, was moved to `CommandRequestImpl` class to use from both `SubscribeButton/UnsubscribeButton`.
- Added logs and minor changes in `CheckHMICapabilities()` function
- Added checking HMI capabilities for `UnsubscribeButton`
- Added new UT for covering new test case and fixes expectations in existing UT's

Related PR: #1358 
Fixes #998

### CODE REVIEW

**Comment by AKalinich-Luxoft** [comment]
@AByzhynar, @LitvinenkoIra, @LuxoftAKutsan, @dev-gh please review

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 1]
```diff
@@ -193,6 +193,12 @@ class CommandRequestImpl : public CommandImpl,
   bool CheckAllowedParameters();
 
   /**
+   * @brief Checks HMI capabilities for specified button support
+   * @param button Button to check
+   */
+  bool CheckHMICapabilities(mobile_apis::ButtonName::eType button) const;
```
Comment (dev-gh) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft 
Perhaps it worth to add 'const' to parameter just for better style.

Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @AKalinich-Luxoft document return value

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 2]
```diff
@@ -550,6 +550,44 @@ bool CommandRequestImpl::CheckAllowedParameters() {
   return true;
 }
 
+bool CommandRequestImpl::CheckHMICapabilities(
+    mobile_apis::ButtonName::eType button) const {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @AKalinich-Luxoft can be const?

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 1 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @dev-gh added in https://github.com/smartdevicelink/sdl_core/pull/1621/commits/e5fb4510b64635d824bbb48a205b7c3e4ed0e92c

Comment (AKalinich-Luxoft) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan added in https://github.com/smartdevicelink/sdl_core/pull/1621/commits/e5fb4510b64635d824bbb48a205b7c3e4ed0e92c

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 2 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan yes, added `const` in https://github.com/smartdevicelink/sdl_core/pull/1621/commits/e5fb4510b64635d824bbb48a205b7c3e4ed0e92c

**Pull request merged**